sscanss/core/io/reader.py

```diff
@@ -595,3 +599,178 @@ def read_robot_world_calibration_file(filename):
+                        # not conforming to Nexus standard (returns array of string not string(NX_char))
+            break
+
+        else:
+            raise AttributeError('There is no NX_class in this file')
+
+        if not data_folder:
+            hdf_interior = hdf_file[main_entry.name]
+            data_folder = main_entry.parent.name
+
+            for _, item in hdf_interior.items():
+                definition = hdf_file.get(f'{item.name}/definition')
+                if definition is None:
+                    continue
+                with suppress(AttributeError):
+                    if definition[()].decode('utf-8').lower() == 'nxtomoproc':
+                        data_folder = definition.parent.name
+                        break
+
+        volume_data = np.array(hdf_file[f'{data_folder}/data/data'])
+        x = np.array(hdf_file[f'{data_folder}/data/x'])
+        y = np.array(hdf_file[f'{data_folder}/data/y'])
+        z = np.array(hdf_file[f'{data_folder}/data/z'])
+        if not (volume_data.shape == (len(x), len(y), len(z))):
+            raise AttributeError('The data arrays in the file are not the same size')
+
+    return Volume(volume_data, x, y, z)
+
+
+def read_single_tiff(filename):
+    """Uses tifffile to open a single TIFF image, returning the result as a numpy array
+    :param filename: filename of the file to open
+    :type filename: str
+    :return: numpy array
+    :return type: numpy.ndarray
+    """
+    image = tiff.imread(str(filename))
+
+    return np.array(image)
+
+
+def file_walker(filepath, extension=(".tiff", ".tif")):
+    """Returns a list of filenames, which satisfy the extension, in the filepath folder
+    :param filepath: path of the folder containing TIFF tiles
+    :type filepath: str
+    :param extension: Tuple of extensions which are searched for
+    :type extension: Tuple[str] or str
+    :return: list of files which have appropriate file extension
+    :return type: array
+    """
+    list_of_files = []
+    for file in os.listdir(filepath):
+        if file.lower().endswith(extension):
+            filename = os.path.join(filepath, file)
+            list_of_files.append(filename)
+
+    return list_of_files
+
+
+def check_tiff_file_size_vs_memory(filepath, instances):
+    """Checks expected size of tiff files in memory and returns False if this exceeds the total free system memory
+    :param filepath: filepath of a single TIFF file
+    :type filepath: str
+    :param instances: number of TIFF files to be loaded
+    :type instances: int
+    :return: Whether the expected size of the file in memory exceeds the available system memory (RAM)
+    :return type: Bool
+    """
+    single_image = read_single_tiff(filepath)
+    size = single_image.nbytes
+    total_size = size * instances
+    should_load = False if total_size >= psutil.virtual_memory().available else True
```

Comment:
Not critical but I just wanted to mention `should_load` is not a good variable name.